# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1408441266 14400
#      Tue Aug 19 05:41:06 2014 -0400
# Node ID d902542221c883b2fb862dceb5bd1d8aeba42b8e
# Parent  1f36981ab323640e95cf6b5fbbfc0df3e42ffd5f
new data directory for storing data files we distribute

* configure.ac (octdatadir): New variable.
* common.mk (do_subst_default_vals): Also substitute octdatadir for
%OCTAVE_OCTDATADIR%.
* defaults.in.h, defaults.cc (OCTAVE_OCTDATADIR): New macro.
(Voct_data_dir): New global variable.
(set_default_oct_data_dir): New static function.
(install_defaults): Call it.
* load-path.cc (load_path::do_initialize): Add Voct_data_dir to the
default system load path.
* toplev.cc (Foctave_config_info): Include octdatadir in result.
* octave-config.in.cc (help_msg, initialize): Include OCTDATADIR in
the list of variables.
* run-octave.in: Include $top_srcdir/etc/data in LOADPATH.

diff --git a/build-aux/common.mk b/build-aux/common.mk
--- a/build-aux/common.mk
+++ b/build-aux/common.mk
@@ -684,16 +684,17 @@ echo "making $@ from $<"
   -e "s|%OCTAVE_LOCALSTARTUPFILEDIR%|\"${localstartupfiledir}\"|" \
   -e "s|%OCTAVE_LOCALAPIARCHLIBDIR%|\"${localapiarchlibdir}\"|" \
   -e "s|%OCTAVE_LOCALVERARCHLIBDIR%|\"${localverarchlibdir}\"|" \
   -e "s|%OCTAVE_LOCALVERFCNFILEDIR%|\"${localverfcnfiledir}\"|" \
   -e "s|%OCTAVE_LOCALVEROCTFILEDIR%|\"${localveroctfiledir}\"|" \
   -e "s|%OCTAVE_MAN1DIR%|\"${man1dir}\"|" \
   -e "s|%OCTAVE_MAN1EXT%|\"${man1ext}\"|" \
   -e "s|%OCTAVE_MANDIR%|\"${mandir}\"|" \
+  -e "s|%OCTAVE_OCTDATADIR%|\"${octdatadir}\"|" \
   -e "s|%OCTAVE_OCTFILEDIR%|\"${octfiledir}\"|" \
   -e "s|%OCTAVE_OCTETCDIR%|\"${octetcdir}\"|" \
   -e "s|%OCTAVE_OCTINCLUDEDIR%|\"${octincludedir}\"|" \
   -e "s|%OCTAVE_OCTLIBDIR%|\"${octlibdir}\"|" \
   -e "s|%OCTAVE_OCTLOCALEDIR%|\"${octlocaledir}\"|" \
   -e "s|%OCTAVE_OCTTESTSDIR%|\"${octtestsdir}\"|" \
   -e "s|%OCTAVE_STARTUPFILEDIR%|\"${startupfiledir}\"|" \
   -e "s|%OCTAVE_PREFIX%|\"${prefix}\"|" \
diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -108,16 +108,17 @@ OCTAVE_SET_DEFAULT([localapifcnfiledir],
   '$(datadir)/octave/site/$(api_version)/m')
 OCTAVE_SET_DEFAULT([localverfcnfiledir], '$(datadir)/octave/$(version)/site/m')
 OCTAVE_SET_DEFAULT([octetcdir], '$(datadir)/octave/$(version)/etc')
 OCTAVE_SET_DEFAULT([octlocaledir], '$(datadir)/octave/$(version)/locale')
 OCTAVE_SET_DEFAULT([doc_cache_file], '$(octetcdir)/doc-cache')
 OCTAVE_SET_DEFAULT([octtestsdir], '$(octetcdir)/tests')
 OCTAVE_SET_DEFAULT([texi_macros_file], '$(octetcdir)/macros.texi')
 OCTAVE_SET_DEFAULT([imagedir], '$(datadir)/octave/$(version)/imagelib')
+OCTAVE_SET_DEFAULT([octdatadir], '$(datadir)/octave/$(version)/data')
 OCTAVE_SET_DEFAULT([man1dir], '$(mandir)/man1')
 OCTAVE_SET_DEFAULT([man1ext], '.1')
 OCTAVE_SET_DEFAULT([infofile], '$(infodir)/octave.info')
 
 ### Check for programs used in building, installing, and running Octave.
 
 ## Programs used in configuring Octave.
 ## Find pkg-config executable (sets $PKG_CONFIG)
diff --git a/libinterp/corefcn/defaults.cc b/libinterp/corefcn/defaults.cc
--- a/libinterp/corefcn/defaults.cc
+++ b/libinterp/corefcn/defaults.cc
@@ -69,16 +69,17 @@ std::string Vlocal_ver_arch_lib_dir;
 std::string Vlocal_ver_oct_file_dir;
 std::string Vlocal_api_oct_file_dir;
 std::string Vlocal_oct_file_dir;
 
 std::string Vlocal_ver_fcn_file_dir;
 std::string Vlocal_api_fcn_file_dir;
 std::string Vlocal_fcn_file_dir;
 
+std::string Voct_data_dir;
 std::string Voct_etc_dir;
 std::string Voct_locale_dir;
 
 std::string Voct_file_dir;
 std::string Vfcn_file_dir;
 
 std::string Vimage_dir;
 
@@ -214,16 +215,22 @@ set_default_fcn_file_dir (void)
 
 static void
 set_default_image_dir (void)
 {
   Vimage_dir = subst_octave_home (OCTAVE_IMAGEDIR);
 }
 
 static void
+set_default_oct_data_dir (void)
+{
+  Voct_data_dir = subst_octave_home (OCTAVE_OCTDATADIR);
+}
+
+static void
 set_default_oct_etc_dir (void)
 {
   Voct_etc_dir = subst_octave_home (OCTAVE_OCTETCDIR);
 }
 
 static void
 set_default_oct_locale_dir (void)
 {
@@ -430,16 +437,17 @@ install_defaults (void)
   set_default_local_ver_oct_file_dir ();
   set_default_local_api_oct_file_dir ();
   set_default_local_oct_file_dir ();
 
   set_default_local_ver_fcn_file_dir ();
   set_default_local_api_fcn_file_dir ();
   set_default_local_fcn_file_dir ();
 
+  set_default_oct_data_dir ();
   set_default_oct_etc_dir ();
   set_default_oct_locale_dir ();
 
   set_default_fcn_file_dir ();
   set_default_oct_file_dir ();
 
   set_default_image_dir ();
 
diff --git a/libinterp/corefcn/defaults.in.h b/libinterp/corefcn/defaults.in.h
--- a/libinterp/corefcn/defaults.in.h
+++ b/libinterp/corefcn/defaults.in.h
@@ -143,16 +143,20 @@ along with Octave; see the file COPYING.
 #ifndef OCTAVE_MAN1EXT
 #define OCTAVE_MAN1EXT %OCTAVE_MAN1EXT%
 #endif
 
 #ifndef OCTAVE_MANDIR
 #define OCTAVE_MANDIR %OCTAVE_MANDIR%
 #endif
 
+#ifndef OCTAVE_OCTDATADIR
+#define OCTAVE_OCTDATADIR %OCTAVE_OCTDATADIR%
+#endif
+
 #ifndef OCTAVE_OCTFILEDIR
 #define OCTAVE_OCTFILEDIR %OCTAVE_OCTFILEDIR%
 #endif
 
 #ifndef OCTAVE_OCTETCDIR
 #define OCTAVE_OCTETCDIR %OCTAVE_OCTETCDIR%
 #endif
 
@@ -197,16 +201,17 @@ extern OCTINTERP_API std::string Vlocal_
 extern OCTINTERP_API std::string Vlocal_ver_oct_file_dir;
 extern OCTINTERP_API std::string Vlocal_api_oct_file_dir;
 extern OCTINTERP_API std::string Vlocal_oct_file_dir;
 
 extern OCTINTERP_API std::string Vlocal_ver_fcn_file_dir;
 extern OCTINTERP_API std::string Vlocal_api_fcn_file_dir;
 extern OCTINTERP_API std::string Vlocal_fcn_file_dir;
 
+extern OCTINTERP_API std::string Voct_data_dir;
 extern OCTINTERP_API std::string Voct_etc_dir;
 extern OCTINTERP_API std::string Voct_locale_dir;
 
 extern OCTINTERP_API std::string Voct_file_dir;
 extern OCTINTERP_API std::string Vfcn_file_dir;
 
 extern OCTINTERP_API std::string Vimage_dir;
 
diff --git a/libinterp/corefcn/load-path.cc b/libinterp/corefcn/load-path.cc
--- a/libinterp/corefcn/load-path.cc
+++ b/libinterp/corefcn/load-path.cc
@@ -564,16 +564,17 @@ load_path::do_initialize (bool set_initi
       maybe_add_path_elts (sys_path, Vlocal_ver_oct_file_dir);
       maybe_add_path_elts (sys_path, Vlocal_api_oct_file_dir);
       maybe_add_path_elts (sys_path, Vlocal_oct_file_dir);
       maybe_add_path_elts (sys_path, Vlocal_ver_fcn_file_dir);
       maybe_add_path_elts (sys_path, Vlocal_api_fcn_file_dir);
       maybe_add_path_elts (sys_path, Vlocal_fcn_file_dir);
       maybe_add_path_elts (sys_path, Voct_file_dir);
       maybe_add_path_elts (sys_path, Vfcn_file_dir);
+      maybe_add_path_elts (sys_path, Voct_data_dir);
     }
 
   std::string tpath = load_path::command_line_path;
 
   if (tpath.empty ())
     tpath = octave_env::getenv ("OCTAVE_PATH");
 
   std::string xpath;
diff --git a/libinterp/corefcn/toplev.cc b/libinterp/corefcn/toplev.cc
--- a/libinterp/corefcn/toplev.cc
+++ b/libinterp/corefcn/toplev.cc
@@ -1423,16 +1423,17 @@ specified option.\n\
     { true, "localoctfiledir", OCTAVE_LOCALOCTFILEDIR },
     { true, "localstartupfiledir", OCTAVE_LOCALSTARTUPFILEDIR },
     { true, "localverarchlibdir", OCTAVE_LOCALVERARCHLIBDIR },
     { true, "localverfcnfiledir", OCTAVE_LOCALVERFCNFILEDIR },
     { true, "localveroctfiledir", OCTAVE_LOCALVEROCTFILEDIR },
     { true, "man1dir", OCTAVE_MAN1DIR },
     { false, "man1ext", OCTAVE_MAN1EXT },
     { true, "mandir", OCTAVE_MANDIR },
+    { true, "octdatadir", OCTAVE_OCTDATADIR },
     { true, "octfiledir", OCTAVE_OCTFILEDIR },
     { true, "octetcdir", OCTAVE_OCTETCDIR },
     { true, "octincludedir", OCTAVE_OCTINCLUDEDIR },
     { true, "octlibdir", OCTAVE_OCTLIBDIR },
     { true, "octtestsdir", OCTAVE_OCTTESTSDIR },
     { true, "prefix", OCTAVE_PREFIX },
     { true, "startupfiledir", OCTAVE_STARTUPFILEDIR },
     { false, "version", OCTAVE_VERSION },
diff --git a/run-octave.in b/run-octave.in
--- a/run-octave.in
+++ b/run-octave.in
@@ -28,28 +28,31 @@ SED=%SED%
 # in these names? 
 
 top_srcdir='%abs_top_srcdir%'
 builddir='%builddir%'
 
 d1="$top_srcdir/scripts"
 d2="$builddir/scripts"
 d3="$builddir/libinterp"
+d4="$top_srcdir/examples/data"
 
 d1_list=`$FIND "$d1" -type d -a ! \( \( -name private -o -name '@*' \) -a -prune \) -exec echo '{}' ';' | $SED 's/$/:/'`
 d2_list=`$FIND "$d2" -type d -a ! \( \( -name private -o -name '@*' \) -a -prune \) -exec echo '{}' ';' | $SED 's/$/:/'`
 d3_list=`$FIND "$d3" -type d -a ! \( \( -name private -o -name '@*' \) -a -prune \) -exec echo '{}' ';' | $SED 's/$/:/'`
+d4_list=`$FIND "$d4" -type d -exec echo '{}' ';' | $SED 's/$/:/'`
 
 d1_path=`echo "$d1_list" | $AWK '{ t = (s $0); s = t; } END { sub (/:$/, "", s); print s; }'`
 d2_path=`echo "$d2_list" | $AWK '{ t = (s $0); s = t; } END { sub (/:$/, "", s); print s; }'`
 d3_path=`echo "$d3_list" | $AWK '{ t = (s $0); s = t; } END { sub (/:$/, "", s); print s; }'`
+d4_path=`echo "$d4_list" | $AWK '{ t = (s $0); s = t; } END { sub (/:$/, "", s); print s; }'`
 
 octave_executable="$builddir/src/octave"
 
-LOADPATH="$d1_path:$d2_path:$d3_path"
+LOADPATH="$d1_path:$d2_path:$d3_path:$d4_path"
 IMAGEPATH="$top_srcdir/scripts/image"
 DOCFILE="$builddir/doc/interpreter/doc-cache"
 BUILT_IN_DOCSTRINGS_FILE="$builddir/libinterp/DOCSTRINGS"
 TEXIMACROSFILE="$top_srcdir/doc/interpreter/macros.texi"
 INFOFILE="$top_srcdir/doc/interpreter/octave.info"
 
 ## Checking for string equality below with prepended x's in order to
 ## handle problems with empty strings.
diff --git a/src/octave-config.in.cc b/src/octave-config.in.cc
--- a/src/octave-config.in.cc
+++ b/src/octave-config.in.cc
@@ -51,33 +51,34 @@ static std::string help_msg =
 "                        expects to find locally installed .m files.\n"
 "\n"
 "  --oct-site-dir        Print the name of the directory where Octave\n"
 "                        expects to find locally installed .oct files.\n"
 "\n"
 "  -p VAR, --print VAR   Print the value of the given configuration\n"
 "                        variable VAR.  Recognized variables are:\n"
 "\n"
-"                          API_VERSION            LOCALAPIOCTFILEDIR\n"
-"                          ARCHLIBDIR             LOCALARCHLIBDIR\n"
-"                          BINDIR                 LOCALFCNFILEDIR\n"
-"                          CANONICAL_HOST_TYPE    LOCALOCTFILEDIR\n"
-"                          DATADIR                LOCALSTARTUPFILEDIR\n"
-"                          DATAROOTDIR            LOCALVERARCHLIBDIR\n"
-"                          DEFAULT_PAGER          LOCALVERFCNFILEDIR\n"
-"                          EXEC_PREFIX            LOCALVEROCTFILEDIR\n"
-"                          FCNFILEDIR             MAN1DIR\n"
-"                          IMAGEDIR               MAN1EXT\n"
-"                          INCLUDEDIR             MANDIR\n"
+"                          API_VERSION            LOCALARCHLIBDIR\n"
+"                          ARCHLIBDIR             LOCALFCNFILEDIR\n"
+"                          BINDIR                 LOCALOCTFILEDIR\n"
+"                          CANONICAL_HOST_TYPE    LOCALSTARTUPFILEDIR\n"
+"                          DATADIR                LOCALVERARCHLIBDIR\n"
+"                          DATAROOTDIR            LOCALVERFCNFILEDIR\n"
+"                          DEFAULT_PAGER          LOCALVEROCTFILEDIR\n"
+"                          EXEC_PREFIX            MAN1DIR\n"
+"                          FCNFILEDIR             MAN1EXT\n"
+"                          IMAGEDIR               MANDIR\n"
+"                          INCLUDEDIR             OCTDATADIR\n"
 "                          INFODIR                OCTFILEDIR\n"
 "                          INFOFILE               OCTINCLUDEDIR\n"
 "                          LIBDIR                 OCTLIBDIR\n"
 "                          LIBEXECDIR             PREFIX\n"
 "                          LOCALAPIARCHLIBDIR     STARTUPFILEDIR\n"
 "                          LOCALAPIFCNFILEDIR     VERSION\n"
+"                          LOCALAPIOCTFILEDIR\n"
 "\n"
 "  -v, --version         Print the Octave version number.\n"
 "\n";
 
 static void
 initialize (void)
 {
   vars["OCTAVE_HOME"] = get_octave_home ();
@@ -108,16 +109,17 @@ initialize (void)
   vars["LOCALFCNFILEDIR"] = subst_octave_home (%OCTAVE_LOCALFCNFILEDIR%);
   vars["LOCALOCTFILEDIR"] = subst_octave_home (%OCTAVE_LOCALOCTFILEDIR%);
   vars["LOCALSTARTUPFILEDIR"] = subst_octave_home (%OCTAVE_LOCALSTARTUPFILEDIR%);
   vars["LOCALVERARCHLIBDIR"] = subst_octave_home (%OCTAVE_LOCALVERARCHLIBDIR%);
   vars["LOCALVERFCNFILEDIR"] = subst_octave_home (%OCTAVE_LOCALVERFCNFILEDIR%);
   vars["LOCALVEROCTFILEDIR"] = subst_octave_home (%OCTAVE_LOCALVEROCTFILEDIR%);
   vars["MAN1DIR"] = subst_octave_home (%OCTAVE_MAN1DIR%);
   vars["MANDIR"] = subst_octave_home (%OCTAVE_MANDIR%);
+  vars["OCTDATADIR"] = subst_octave_home (%OCTAVE_OCTDATADIR%);
   vars["OCTFILEDIR"] = subst_octave_home (%OCTAVE_OCTFILEDIR%);
   vars["OCTINCLUDEDIR"] = subst_octave_home (%OCTAVE_OCTINCLUDEDIR%);
   vars["OCTLIBDIR"] = subst_octave_home (%OCTAVE_OCTLIBDIR%);
   vars["STARTUPFILEDIR"] = subst_octave_home (%OCTAVE_STARTUPFILEDIR%);
 }
 
 int
 main (int argc, char **argv)
